lib/events/filelog.go

```diff
@@ -230,6 +308,17 @@ func (l *FileLog) searchEventsWithFilter(fromUTC, toUTC time.Time, limit int, or
 		return nil, "", trace.BadParameter("limit %v exceeds max iteration limit %v", limit, defaults.MaxIterationLimit)
 	}
 
+	var checkpoint checkpointKey
+	if err := checkpoint.UnmarshalText([]byte(startAfter)); err != nil {
+		return nil, "", trace.Wrap(err, "parsing startAfter")
+	}
+
+	// If the checkpoint is before the requested timeframe, we knoe that every
+	// event we'll read is valid. We must unset the checkpoint to avoid returning no event.
```

Comment:
Should we also immediately return no events if the checkpoint is after the requested time frame?